src/basics.ml

```diff
@@ -216,3 +216,53 @@ let term_of_rhs : rule -> term = fun r ->
     TE_Some(Bindlib.unbox (Bindlib.bind_mvar vars p))
   in
   Bindlib.msubst r.rhs (Array.mapi fn r.vars)
+
+(** {3 Subterm position helper module} *)
+
+(** Some tools to encode the position of a subterm in a term. *)
+module Subterm =
+struct
+  (** Each element of the list is a level in the tree of the term.  For
+      instance, in the term [Appl(Appl(f, x), Appl(Appl(g, a), b))], the
+      subterm [a] has position [1.0], encoded by [[0 ; 1]], [b] has [1.1]
+      encoded by [[1 ; 1]] and [x] has [0] encoded by [[0]]. *)
+  type t = int list
+
+  (** [compare a b] implements lexicographic order on positions. *)
+  let compare : t -> t -> int = fun a b -> compare (List.rev a) (List.rev b)
```

Comment:
Use Pervasives.compare.

Reply:
You mean removing the `List.rev` or replacing `compare` by `Pervasives.compare`?

Reply:
replacing compare by Pervasives.compare